### DISCUSSION TITLE
Consider using range library

### DISCUSSION THREAD

**Discussion by scarf005** [cataclysmbn/Cataclysm-BN#1996] [General]
codebase need better way to manage ranges, something like...
```c++
for ( int i : range( v.size( ) ) ) { 
  for ( int j : range( -3, 3 + 1 ) ) { 
    // ...
  }
} 

for ( auto&& [ i, value ] : enumerate( v ) ) {
  //  ...
}
```
### implement a own. 
or:
### https://github.com/klmr/cpp11-range
pros: small, single header, performant
cons: only has `ranges`

### https://github.com/ryanhaining/cppitertools
pros: python-like syntax, tons of itertools to clean up code
cons: heavy (lots of headers), official build tool only supports bevy and scons

### C++20 ranges
pros: C++ standard.
cons: need to migrate to C++20, #1624 is WIP

### https://github.com/ericniebler/range-v3
pros: superset of C++20 ranges library
cons: long compile time

**Comment by Coolthulhu** [comment]
I'd say C++20 or our own. I'd rather not add more libraries for now.
